rust_snuba/rust_arroyo/src/processing/dlq.rs

```diff
@@ -493,6 +495,38 @@ mod tests {
         assert_eq!(*producer.call_count.lock().unwrap(), 10);
     }
 
+    #[test]
+    #[should_panic]
```

Comment:
i'd prefer to assert that a specific line below panics rather than the whole function, but it doesn't work because `UnwindSafe` is not implemented for `DlqPolicyWrapper`

Reply:
do you have a piece of code that you tried? it might be because you are accessing DlqPolicyWrapper after the panic. if you are dropping it before it might work